OpenFeature/src/main/java/dev/openfeature/sdk/events/EventHandler.kt

```diff
@@ -2,29 +2,28 @@ package dev.openfeature.sdk.events
 
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.Job
 import kotlinx.coroutines.cancelChildren
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.filterIsInstance
 import kotlinx.coroutines.launch
-import kotlin.reflect.KClass
 
-interface ProviderStatus {
-    fun isProviderReady(): Boolean
+interface EventObserver {
+    fun observe(): Flow<OpenFeatureEvents>
 }
 
-interface EventObserver {
-    fun <T : OpenFeatureEvents> observe(kClass: KClass<T>): Flow<T>
+interface ProviderStatus {
```

Comment:
nit: I would expect an interface such as this to have a `getStatus(): Status` function.